executor/metric_reader.go

```diff
@@ -62,9 +63,14 @@ func (e *MetricRetriever) retrieve(ctx context.Context, sctx sessionctx.Context)
 		quantiles = []float64{tblDef.Quantile}
 	}
 	for _, quantile := range quantiles {
-		queryValue, err := e.queryMetric(ctx, sctx, queryRange, quantile)
-		if err != nil {
-			return nil, err
+		var queryValue pmodel.Value
+		// Add retry to avoid network error.
+		for i := 0; i < 10; i++ {
+			queryValue, err = e.queryMetric(ctx, sctx, queryRange, quantile)
+			if err == nil {
+				break
+			}
+			time.Sleep(100 * time.Millisecond)
```

Comment:
Check if it's a network error before continuing.
E.g. it may be a parse error, then it will retry 10 times.

Reply:
great, done.